src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectAssetFileWatcher.cs

```diff
@@ -50,9 +51,27 @@ internal class ProjectAssetFileWatcher : OnceInitializedOnceDisposed, IVsFileCha
         /// </summary>
         [ConfiguredProjectAutoLoad(RequiresUIThread = true)]
         [AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharp)]
-        internal void Load()
+        internal async Task LoadAsync()
         {
             EnsureInitialized();
+
+            await Task.Run(() =>
```

Comment:
The pattern we use elsewhere is;

- Use OnceInitializedOnceDisposed*Async*
- await TaskScheduler.Default to switch back to the background thread, such http://index/?leftProject=mscorlib&leftSymbol=j8ur3kjbcsbf&rightProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&file=Build%5CBuildManagerAccessorDesignTime.cs&line=300. Not sure what namespace/dll adds the awaiter to the TaskScheduler.

Alternatively:

- Don't force UI thread
- Use IAsyncServiceProvider (note, it doesn't come from MEF yet so you need to use this pattern: https://github.com/dotnet/project-system/blob/af97c6b945847850831e9ca9d1ade2d4038788c3/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs#L40).

Reply:
Ha, but looks like OnceInitializedOnceDisposedAsync inits on a background thread, so the first one wouldn't work.

Reply:
For future reference the awaiter comes from this namespace/type: https://github.com/Microsoft/vs-threading/blob/6d35763321204bde522a58d127d670141d83fefb/src/Microsoft.VisualStudio.Threading/AwaitExtensions.cs#L32.

Reply:
Also you don't need the async machine here, you can just return Task.Run.